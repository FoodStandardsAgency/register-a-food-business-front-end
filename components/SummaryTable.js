const SummaryTable = props => (
  <div>
    {props.establishment_first_line ? (
      <div id="establishmentAddress">
        <p>Establishment address:</p>
        <p>{props.establishment_first_line}</p>
      </div>
    ) : null}
  </div>
);

export default SummaryTable;
