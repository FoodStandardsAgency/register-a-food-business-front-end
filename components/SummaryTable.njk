{% from "govuk/components/inset-text/macro.njk" import govukInsetText %}
{% from "govuk/components/button/macro.njk" import govukButton %}
{% from "../components/PostForm.njk" import postForm %}
{% from "../components/ProcessedErrorSummary.njk" import processedErrorSummary %}
{% from "../components/SummaryListHelper.njk" import summaryListHelperWithoutEdit, summaryListHelperHtmlWithoutEdit, summaryListDeclarationHelper %}
{% macro summaryTable(props) %}
  {% set propsData = props.transformedData %}
  {# {% set propsData.applicationCompletePage = true %} #}
  {% set validatorErrors = props.allValidationErrors %}
  <fieldset class="govuk-fieldset">
    <h2 id="operator-details" class="govuk-heading-m">{{ __("Operator details", props.language) }}</h2>
    {% if propsData.applicationCompletePage == null %}
      {% if validatorErrors.operator_type or propsData.operator_type %}
        {{ summaryListHelperWithoutEdit(__("Operator type", props.language), __(propsData.operator_type, props.language), false) }}
      {% endif %}
    {% endif %}
    {% if propsData.applicationCompletePage == null %}
      {% if validatorErrors.operator_company_name or propsData.operator_company_name %}
        {{ summaryListHelperWithoutEdit(__("Company name", props.language), propsData.operator_company_name, false ) }}
      {% endif %}
    {% endif %}
    {% if propsData.applicationCompletePage == null %}
      {% if validatorErrors.companies_house_number or propsData.operator_companies_house_number %}
        {{ summaryListHelperWithoutEdit(__("Companies House number", props.language), propsData.operator_companies_house_number, false ) }}
      {% endif %}
    {% endif %}
    {% if propsData.applicationCompletePage == null %}
      {% if validatorErrors.operator_charity_name or propsData.operator_charity_name %}
        {{ summaryListHelperWithoutEdit(__("Charity name", props.language), propsData.operator_charity_name, false ) }}
      {% endif %}
    {% endif %}
    {% if propsData.applicationCompletePage == null %}
      {% if validatorErrors.operator_charity_number or propsData.operator_charity_number %}
        {{ summaryListHelperWithoutEdit(__("Charity number", props.language), propsData.operator_charity_number, false) }}
      {% endif %}
    {% endif %}
    {% if propsData.applicationCompletePage == null %}
      {% if validatorErrors.operator_first_name or propsData.operator_first_name %}
        {{ summaryListHelperWithoutEdit(__("Name", props.language), propsData.operator_first_name + " " + propsData.operator_last_name, false ) }}
      {% endif %}
    {% endif %}
    {% if propsData.applicationCompletePage == null %}
      {% if validatorErrors.operator_address_line_1 or validatorErrors.operator_address_line_2 or validatorErrors.operator_address_line_3 or 
                    validatorErrors.operator.operator_town or validatorErrors.operator.operator_postcode or validatorErrors.operator_postcode_find or 
                    propsData.operator_address_line_1 or propsData.operator_address_line_2 or propsData.operator_address_line_3 or propsData.operator_town or propsData.operator_postcode or propsData.operator_postcode_find %}
        {% set htmlValue = (propsData.operator_address_line_1 + "<br>" + if propsData.operator_address_line_1) + (propsData.operator_address_line_2 + "<br>" + if propsData.operator_address_line_2) + (propsData.operator_address_line_3 + "<br>" + if propsData.operator_address_line_3)+ propsData.operator_town + "<br>" + propsData.operator_postcode %}
        {{ summaryListHelperHtmlWithoutEdit(__("Partnership contact address", props.language) if  propsData.operator_type === "PARTNERSHIP" else __("Operator address", props.language), htmlValue, false) }}
      {% endif %}
    {% endif %}
    {% if propsData.applicationCompletePage == null %}
      {% if validatorErrors.operator_primary_number or validatorErrors.operator_secondary_number or propsData.operator_primary_number %}
        {{ summaryListHelperHtmlWithoutEdit(__("Phone number", props.language), propsData.operator_primary_number + "<br>" + propsData.operator_secondary_number, false ) }}
      {% endif %}
    {% endif %}
    {% if propsData.applicationCompletePage == null %}
      {% if validatorErrors.operator_email or propsData.operator_email %}
        {{ summaryListHelperWithoutEdit(__("Email address", props.language), propsData.operator_email, false) }}
      {% endif %}
    {% endif %}
    {% if propsData.applicationCompletePage == null %}
      {% if validatorErrors.main_partnership_contact or propsData.main_partnership_contact %}
        {{ summaryListHelperWithoutEdit(__("Main partnership contact", props.language), propsData.main_partnership_contact, false) }}
      {% endif %}
    {% endif %}
    {% if propsData.applicationCompletePage == null %}
      {% if validatorErrors.partners or propsData.partners %}
        {{ summaryListHelperWithoutEdit(__("Partners", props.language), propsData.partners, false) }}
      {% endif %}
    {% endif %}
    {% if propsData.applicationCompletePage == null %}
      {% if validatorErrors.contact_representative_name or validatorErrors.contact_representative_role or validatorErrors.contact_representative_number or validatorErrors.contact_representative_email 
                    or propsData.contact_representative_name or propsData.contact_representative_role or propsData.contact_representative_number or propsData.contact_representative_email %}
        {% set contactRepRole = propsData.contact_representative_role if propsData.contact_representative_role else "" %}
        {% set htmlValue = (propsData.contact_representative_name if propsData.contact_representative_name else "") + "<br>" 
                                + (propsData.contact_representative_role if propsData.contact_representative_role else "") + "<br>" 
                                + (propsData.contact_representative_number if propsData.contact_representative_number else "") + "<br>" 
                                + (propsData.contact_representative_email if propsData.contact_representative_email else "") %}
        {{ summaryListHelperHtmlWithoutEdit(__("Designated contact", props.language), htmlValue, false) }}
      {% endif %}
    {% endif %}
    <h2 id="establishment-details" class="govuk-heading-m">{{ __("Establishment details", props.language) }}</h2>
    {% if propsData.applicationCompletePage == null %}
      {% if validatorErrors.establishment_trading_name or propsData.establishment_trading_name %}
        {{ summaryListHelperWithoutEdit(__("Trading name", props.language), propsData.establishment_trading_name, false) }}
      {% endif %}
    {% endif %}
    {% if propsData.applicationCompletePage == null %}
      {% if propsData.establishment_address_line_1 or propsData.establishment_postcode_find or validatorErrors.establishment_address_line_1 or validatorErrors.establishment_address_line_2 
                    or  validatorErrors.establishment_address_line_3 or validatorErrors.establishment_town or  validatorErrors.establishment_postcode or validatorErrors.establishment_postcode_find %}
        {% set htmlValue = (propsData.establishment_address_line_1 + "<br>" if propsData.establishment_address_line_1) + (propsData.establishment_address_line_2 + "<br>" if propsData.establishment_address_line_2) + (propsData.establishment_address_line_3 + "<br>" if propsData.establishment_address_line_3) + propsData.establishment_town + "<br>" + propsData.establishment_postcode %}
        {{ summaryListHelperHtmlWithoutEdit(__("Establishment address", props.language), htmlValue, false ) }}
      {% endif %}
    {% endif %}
    {% if propsData.applicationCompletePage == null %}
      {% if validatorErrors.establishment_type or propsData.establishment_type %}
        {{ summaryListHelperWithoutEdit(__("Address type", props.language), __(propsData.establishment_type, props.language), false) }}
      {% endif %}
    {% endif %}
    {% if propsData.applicationCompletePage == null %}
      {% if validatorErrors.establishment_primary_number or validatorErrors.establishment_secondary_number or propsData.establishment_primary_number %}
        {{ summaryListHelperHtmlWithoutEdit(__("Phone number", props.language), propsData.establishment_primary_number + "<br>" + propsData.establishment_secondary_number, false ) }}
      {% endif %}
    {% endif %}
    {% if propsData.applicationCompletePage == null %}
      {% if validatorErrors.establishment_email or propsData.establishment_email %}
        {{ summaryListHelperWithoutEdit(__("Email address", props.language), propsData.establishment_email, false, false) }}
      {% endif %}
    {% endif %}
    {% if propsData.applicationCompletePage == null %}
      {% if validatorErrors.establishment_web_address or propsData.establishment_web_address %}
        {{ summaryListHelperHtmlWithoutEdit(__("Web address", props.language), propsData.establishment_web_address, false, false) }}
      {% endif %}
    {% endif %}
    {% if propsData.applicationCompletePage == null %}
      {% if propsData.establishment_opening_date or validatorErrors.establishment_opening_status or validatorErrors.establishment_opening_date %}
        {% set htmlValue = propsData.establishment_opening_date | date(props.language, "DD MMM YYYY")  if propsData.establishment_opening_date else "" %}
        {{ summaryListHelperWithoutEdit(__("Trading date", props.language), htmlValue, false) }}
      {% endif %}
    {% endif %}
    {% if propsData.applicationCompletePage == null %}
      {% if propsData.opening_days_start or propsData.opening_days_some or propsData.opening_days_irregular or validatorErrors.opening_days_start or validatorErrors.opening_days_some or validatorErrors.opening_days_irregular %}
        {% set html %}
        {% if propsData.opening_days_irregular == null %}
          <div>
            <div id="opening_day_monday">{{ __(propsData.opening_day_monday, props.language) if propsData.opening_day_monday }}</div>
            <div id="opening_day_tuesday">{{ __(propsData.opening_day_tuesday, props.language) if propsData.opening_day_tuesday }}</div>
            <div id="opening_day_wednesday">{{ __(propsData.opening_day_wednesday, props.language) if propsData.opening_day_wednesday }}</div>
            <div id="opening_day_thursday">{{ __(propsData.opening_day_thursday, props.language) if propsData.opening_day_thursday }}</div>
            <div id="opening_day_friday">{{ __(propsData.opening_day_friday, props.language) if propsData.opening_day_friday }}</div>
            <div id="opening_day_saturday">{{ __(propsData.opening_day_saturday, props.language) if propsData.opening_day_saturday }}</div>
            <div id="opening_day_sunday">{{ __(propsData.opening_day_sunday, props.language) if propsData.opening_day_sunday }}</div>
          </div>
        {% else %}
          {{ propsData.opening_days_irregular }}
        {% endif %}
        {% endset %}
        {{ summaryListHelperHtmlWithoutEdit(__("Opening days", props.language), html, false ) }}
      {% endif %}
    {% endif %}
    {% if propsData.applicationCompletePage == null %}
      {% if (propsData.opening_days_start and propsData.opening_days_start !== "Irregular days") or propsData.opening_days_some or validatorErrors.opening_hours_monday or validatorErrors.opening_hours_tuesday 
                    or validatorErrors.opening_hours_wednesday or validatorErrors.opening_hours_thursday or validatorErrors.opening_hours_friday or validatorErrors.opening_hours_saturday or validatorErrors.opening_hours_sunday %}
        {% set html %}
        <div>
          <div id="opening_hours_monday">{{ __("Monday", props.language) + ": " + propsData.opening_hours_monday if propsData.opening_hours_monday }}</div>
          <div id="opening_hours_tuesday">{{ __("Tuesday", props.language) + ": " + propsData.opening_hours_tuesday if propsData.opening_hours_tuesday }}</div>
          <div id="opening_hours_wednesday">{{ __("Wednesday", props.language) + ": " + propsData.opening_hours_wednesday if propsData.opening_hours_wednesday }}</div>
          <div id="opening_hours_thursday">{{ __("Thursday", props.language) + ": " + propsData.opening_hours_thursday if propsData.opening_hours_thursday }}</div>
          <div id="opening_hours_friday">{{ __("Friday", props.language) + ":" + propsData.opening_hours_friday if propsData.opening_hours_friday }}</div>
          <div id="opening_hours_saturday">{{ __("Saturday", props.language) + ": " + propsData.opening_hours_saturday if propsData.opening_hours_saturday }}</div>
          <div id="opening_hours_sunday">{{ __("Sunday", props.language) + ": " + propsData.opening_hours_sunday if propsData.opening_hours_sunday }}</div>
        </div>
        {% endset %}
        {{ summaryListHelperHtmlWithoutEdit(__("Opening hours", props.language), html, false ) }}
      {% endif %}
    {% endif %}
    {% if propsData.applicationCompletePage == null %}
      {% if validatorErrors.water_supply or propsData.water_supply %}
        {{ summaryListHelperWithoutEdit(__("Water supply", props.language), __(propsData.water_supply, props.language), false) }}
      {% endif %}
    {% endif %}
    <h2 id="activity-details" class="govuk-heading-m">{{ __("Activities", props.language) }}</h2>
    {% if propsData.applicationCompletePage == null %}
      {% if validatorErrors.customer_type or propsData.customer_type %}
        {{ summaryListHelperWithoutEdit(__("Customers", props.language), __(propsData.customer_type, props.language), false) }}
      {% endif %}
    {% endif %}
    {% if propsData.applicationCompletePage == null %}
      {% if validatorErrors.business_type or propsData.business_type %}
        {{ summaryListHelperWithoutEdit(__("Business type", props.language), __(propsData.business_type, props.language), false) }}
      {% endif %}
    {% endif %}
    {% if propsData.applicationCompletePage == null %}
      {% if validatorErrors.import_export_activities or propsData.import_export_activities %}
        {{ summaryListHelperWithoutEdit(__("Import and export", props.language), __(propsData.import_export_activities, props.language), false) }}
      {% endif %}
    {% endif %}
    {% if propsData.applicationCompletePage == null %}
      {% if validatorErrors.business_other_details or propsData.business_other_details %}
        {{ summaryListHelperWithoutEdit(__("Additional details", props.language), propsData.business_other_details, false) }}
      {% endif %}
    {% endif %}
    <h2 id="declaration-details" class="govuk-heading-m">{{ __("Declaration", props.language) }}</h2>
    {% if props.cumulativeFullAnswers.declaration1 %}
      {% set html %}
      <span id="declaration1">
        {{ props.declaration1 }}</span>
      {% endset %}
      {{ summaryListDeclarationHelper(html, __("Accepted", props.language) ) }}
    {% endif %}
    {% if props.cumulativeFullAnswers.declaration2 %}
      {% set html %}
      <span id="declaration2">{{ props.declaration2 }}</span>
      {% endset %}
      {{ summaryListDeclarationHelper(html, __("Accepted", props.language) ) }}
    {% endif %}
    {% if props.cumulativeFullAnswers.declaration2 %}
      {% set html %}
      <span id="declaration3">
        {{ props.declaration3 }}</span>
      {% endset %}
      {{ summaryListDeclarationHelper(html, __("Accepted", props.language) ) }}
    {% endif %}
  </fieldset>
{% endmacro %}