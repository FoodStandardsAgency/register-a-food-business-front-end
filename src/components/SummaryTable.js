import React from "react";
import { Table, Paragraph } from "govuk-react";
import PropTypes from "prop-types";
import * as COLOUR from "govuk-colours";
import styled from "react-emotion";
import {
  FONT_SIZE,
  LINE_HEIGHT,
  MEDIA_QUERIES,
  NTA_LIGHT
} from "@govuk-react/constants";

const StyledTableRow = styled("div")({
  fontFamily: NTA_LIGHT,
  WebkitFontSmoothing: "antialiased",
  MozOsxFontSmoothing: "grayscale",
  fontWeight: 400,
  display: "inline",
  textTransform: "none",
  fontSize: FONT_SIZE.SIZE_16,
  lineHeight: LINE_HEIGHT.SIZE_16,
  [MEDIA_QUERIES.LARGESCREEN]: {
    fontSize: FONT_SIZE.SIZE_19,
    lineHeight: LINE_HEIGHT.SIZE_19
  },
  color: `${COLOUR.GREY_1}`
});

const OperatorDetailsTable = props => (
  <React.Fragment>
    <Table.Row TITLE>
      <Table.CellHeader>
        <Paragraph mb={0}>**Operator details**</Paragraph>
      </Table.CellHeader>
      <Table.Cell>
        <div />
      </Table.Cell>
    </Table.Row>

    {props.operator_company_name ? (
      <Table.Row id="operatorCompanyNameRow">
        <Table.CellHeader>Company name</Table.CellHeader>
        <Table.Cell>
          <div id="operator_company_name" className="bold">
            {props.operator_company_name}
          </div>
        </Table.Cell>
      </Table.Row>
    ) : null}

    {props.operator_company_house_number ? (
      <Table.Row id="operatorCompaniesHouseRow">
        <Table.CellHeader>Company number</Table.CellHeader>
        <Table.Cell>
          <div id="operator_company_house_number" className="bold">
            {props.operator_company_house_number}
          </div>
        </Table.Cell>
      </Table.Row>
    ) : null}

    {props.operator_charity_name ? (
      <Table.Row id="operatorCharityNameRow">
        <Table.CellHeader>Charity name</Table.CellHeader>
        <Table.Cell>
          <div id="operator_charity_name" className="bold">
            {props.operator_charity_name}
          </div>
        </Table.Cell>
      </Table.Row>
    ) : null}

    {props.operator_charity_number ? (
      <Table.Row id="operatorCharityNumberRow">
        <Table.CellHeader>Charity number</Table.CellHeader>
        <Table.Cell>
          <div id="operator_charity_number" className="bold">
            {props.operator_charity_number}
          </div>
        </Table.Cell>
      </Table.Row>
    ) : null}

    {props.operator_first_name ? (
      <Table.Row id="operatorNameRow">
        <Table.CellHeader>Name</Table.CellHeader>
        <Table.Cell>
          <div className="bold">
            <span id="operator_first_name">{props.operator_first_name}</span>{" "}
            <span id="operator_last_name">{props.operator_last_name}</span>
          </div>
        </Table.Cell>
      </Table.Row>
    ) : null}

    {props.operator_first_line ? (
      <Table.Row id="operatorAddressRow">
        <Table.CellHeader>Operator address</Table.CellHeader>
        <Table.Cell>
          <div className="bold">
            <div id="operator_first_line">{props.operator_first_line}</div>
            <div id="operator_street">{props.operator_street || null}</div>
            <div id="operator_town">{props.operator_town || null}</div>
            <div id="operator_postcode">{props.operator_postcode}</div>
          </div>
        </Table.Cell>
      </Table.Row>
    ) : null}

    {props.operator_primary_number ? (
      <Table.Row id="operatorContactDetailsRow">
        <Table.CellHeader>Contact number</Table.CellHeader>
        <Table.Cell>
          <div className="bold">
            <div id="operator_primary_number">
              {props.operator_primary_number}
            </div>
            <div id="operator_secondary_number">
              {props.operator_secondary_number || null}
            </div>
          </div>
        </Table.Cell>
      </Table.Row>
    ) : null}

    {props.operator_type ? (
      <Table.Row id="operatorTypeRow">
        <Table.CellHeader>Operator type</Table.CellHeader>
        <Table.Cell>
          <div className="bold" id="operator_type">
            {props.operator_type}
          </div>
        </Table.Cell>
      </Table.Row>
    ) : null}

    {props.operator_email ? (
      <Table.Row id="operatorEmailRow">
        <Table.CellHeader>Email address</Table.CellHeader>
        <Table.Cell>
          <div className="bold" id="operator_email">
            {props.operator_email}
          </div>
        </Table.Cell>
      </Table.Row>
    ) : null}

    {props.contact_representative_email ? (
      <Table.Row id="contactRepresentativeRow">
        <Table.CellHeader>Email address</Table.CellHeader>
        <Table.Cell>
          <StyledTableRow
            className="contact_representative_name"
            id="contact_representative_name"
          >
            {props.contact_representative_name}{" "}
          </StyledTableRow>
          <StyledTableRow
            className="contact_representative_role"
            id="contact_representative_role"
          >
            {props.contact_representative_role
              ? `\u2022 ${props.contact_representative_role}`
              : null}
          </StyledTableRow>
          <div className="bold" id="contact_representative_name">
            {props.contact_representative_number}
          </div>
          <div className="bold" id="contact_representative_role">
            {props.contact_representative_email}
          </div>
        </Table.Cell>
      </Table.Row>
    ) : null}
  </React.Fragment>
);

const EstablishmentDetailsTable = props => (
  <React.Fragment>
    <Table.Row TITLE>
      <Table.CellHeader>
        <Paragraph mb={0}>**Establishment details**</Paragraph>
      </Table.CellHeader>
      <Table.Cell>
        <div />
      </Table.Cell>
    </Table.Row>

    {props.establishment_trading_name ? (
      <Table.Row id="establishmentTradingNameRow">
        <Table.CellHeader>Establishment name</Table.CellHeader>
        <Table.Cell>
          <div className="bold" id="establishment_trading_name">
            {props.establishment_trading_name}
          </div>
        </Table.Cell>
      </Table.Row>
    ) : null}

    {props.establishment_first_line ? (
      <Table.Row id="establishmentAddressRow">
        <Table.CellHeader>Establishment address</Table.CellHeader>
        <Table.Cell>
          <div className="bold">
            <div id="establishment_first_line">
              {props.establishment_first_line}
            </div>
            <div id="establishment_street">
              {props.establishment_street || null}
            </div>
            <div id="establishment_town">
              {props.establishment_town || null}
            </div>
            <div id="establishment_postcode">
              {props.establishment_postcode}
            </div>
          </div>
        </Table.Cell>
      </Table.Row>
    ) : null}

    {props.establishment_primary_number ? (
      <Table.Row id="establishmentContactDetailsRow">
        <Table.CellHeader>Contact number</Table.CellHeader>
        <Table.Cell>
          <div className="bold">
            <div id="establishment_primary_number">
              {props.establishment_primary_number}
            </div>
            <div id="establishment_secondary_number">
              {props.establishment_secondary_number || null}
            </div>
          </div>
        </Table.Cell>
      </Table.Row>
    ) : null}

    {props.establishment_email ? (
      <Table.Row id="establishmentEmailRow">
        <Table.CellHeader>Email address</Table.CellHeader>
        <Table.Cell>
          <div className="bold" id="establishment_email">
            {props.establishment_email || null}
          </div>
        </Table.Cell>
      </Table.Row>
    ) : null}
  </React.Fragment>
);

const FoodActivitiesTable = props => (
  <React.Fragment>
    <Table.Row TITLE>
      <Table.CellHeader>
        <Paragraph mb={0}>**Food activities**</Paragraph>
      </Table.CellHeader>
      <Table.Cell>
        <div />
      </Table.Cell>
    </Table.Row>

    {props.customer_type ? (
      <Table.Row id="activitiesCustomersRow">
        <Table.CellHeader>Customers</Table.CellHeader>
        <Table.Cell>
          <div className="bold" id="customer_type">
            {props.customer_type}
          </div>
        </Table.Cell>
      </Table.Row>
    ) : null}
  </React.Fragment>
);

const SummaryTable = props => (
  <Table
    body={
      <React.Fragment>
        <OperatorDetailsTable {...props} />
        <EstablishmentDetailsTable {...props} />
        <FoodActivitiesTable {...props} />
      </React.Fragment>
    }
  />
);

export default SummaryTable;

SummaryTable.propTypes = {
  operator_company_name: PropTypes.string,
  operator_company_house_number: PropTypes.string,
  operator_charity_name: PropTypes.string,
  operator_charity_number: PropTypes.string,
  operator_first_line: PropTypes.string,
  operator_street: PropTypes.string,
  operator_town: PropTypes.string,
  operator_postcode: PropTypes.string,
  operator_first_name: PropTypes.string,
  operator_last_name: PropTypes.string,
  registration_role: PropTypes.string,
  establishment_trading_name: PropTypes.string,
  establishment_first_line: PropTypes.string,
  establishment_street: PropTypes.string,
  establishment_town: PropTypes.string,
  establishment_postcode: PropTypes.string,
  customer_type: PropTypes.string
};
