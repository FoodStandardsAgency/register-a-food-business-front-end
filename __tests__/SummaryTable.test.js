import SummaryTable from "../components/SummaryTable";
import { shallow, mount } from "enzyme";
import renderer from "react-test-renderer";
import * as emotion from "emotion";
import { createSerializer } from "jest-emotion";

expect.addSnapshotSerializer(createSerializer(emotion));

const testFullAnswers = { establishment_first_line: "Example first line" };
const testMissingAnswers = {
  not_expected_on_page: "This should never be rendered"
};

describe("<SummaryTable />", () => {
  it("renders without crashing", () => {
    const wrapper = shallow(<SummaryTable />);
    expect(wrapper.length).toBe(1);
  });

  it("matches the previous snapshot", () => {
    const tree = renderer
      .create(<SummaryTable {...testFullAnswers} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("gets given answers as props", () => {
    const wrapper = mount(<SummaryTable {...testFullAnswers} />);
    const props = wrapper.props();
    expect(props).toEqual(testFullAnswers);
  });

  describe("establishment address table row", () => {
    const testEstablishmentAddressAnswers = {
      establishment_first_line: "Example first line"
    };

    it("renders when the establishment_first_line answer is provided", () => {
      const wrapper = mount(
        <SummaryTable {...testEstablishmentAddressAnswers} />
      );
      const row = wrapper.find("div#establishmentAddress");
      expect(row.length).toBe(1);
    });

    it("does not render when the establishment_first_line answer is not provided", () => {
      const wrapper = mount(<SummaryTable {...testMissingAnswers} />);
      const row = wrapper.find("div#establishmentAddress");
      expect(row.length).toBe(0);
    });

    it("contains all of the establishment address information", () => {
      const wrapper = mount(
        <SummaryTable {...testEstablishmentAddressAnswers} />
      );
      // const text = wrapper.find("div#establishmentAddress");

      // expect(text).toBe(0);
    });
  });
});
