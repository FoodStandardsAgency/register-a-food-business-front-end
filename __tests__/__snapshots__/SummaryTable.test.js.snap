// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SummaryTable /> matches the previous snapshot 1`] = `
<div>
  <div
    id="establishmentAddress"
  >
    <p>
      Establishment address:
    </p>
    <p>
      Example first line
    </p>
  </div>
</div>
`;
